src/protocol.c

```diff
@@ -136,8 +138,16 @@ thread_run_command(void *args) {
                 perror("setenv");
                 exit(1);
             }
-            if (execvp(server->argv[0], server->argv) < 0) {
-                perror("execvp");
+            int e = -1;
+            if ( 0 == access(server->argv[0], R_OK | X_OK) ) {
```

Comment:
Maybe we can parse it ourself, I don't know if there's a C library can do this, but I've seen one written in Go.